04_documentation/code_documentation_tools_exercise.md

```diff
@@ -0,0 +1,92 @@
+    - Change the name of the website (`site_name`) from "My Docs" to "SSE Documentation Example"
+    - [Add a navigation section](https://www.mkdocs.org/user-guide/writing-your-docs/#configure-pages-and-navigation) `nav` and add the three pages manually:
+        - Add the `index.md` with label `Home`.
+        - Create a section `User Guide` and add the following subsections
+            - Add the `compilation-and-installation.md` with label `Installation`.
+            - Add the `how-to.md` with label `Getting Started`
+          You should only have two buttons "Home" and "User Guide" in the generated website.
+    - Add the [repository URL](https://www.mkdocs.org/user-guide/configuration/#repo_url). This adds a "Edit on GitHub" button at the top right of the website. Try if this works. If it does not work, this is because older versions of MkDocs set a link to the `master` branch by default, but GitHub calls this branch `main` nowadays. In order to point to the correct branch, set the `edit_uri` setting to `edit/main/docs/`.
+
+        **Note** The "Edit on GitHub" also fails if you have not pushed the corresponding Markdown file to your repository yet.
+
+    - Make yourself the [website's author](https://www.mkdocs.org/user-guide/configuration/#site_author). You can use your real name or your GitHub username. This setting will not change the appearance of your website, but adds the name to the HTML metadata.
+- [Deploy your website](https://www.mkdocs.org/user-guide/deploying-your-docs/) to [GitHub Pages](https://pages.github.com/). This consists of following steps:
+    - Run `mkdocs gh-deploy` to build and deploy the website:
+
+      Deployment of the website will create a branch `gh-pages` inside the repository if it does not already exist. This branch contains all files needed to deliver the website. You should not edit any of the files in the branch manually since `mkdocs gh-deploy` will overwrite all changes. Instead, fix problems in your main branch and deploy the website afterwards again.
+
+    - Verify that the "Pages" feature of your repository has been activated. When you are on your repository on GitHub go to "Settings" (at top of window) -> "Pages". It should show an information box " Your site is ready to be published at URL" with the URL to the website of your documentation.
+
+      If the "Pages" feature has not been activated automatically, choose the `gh-pages` branch as source and save/activate the website.
+
+      Check the website. If you get a "404" error
+
+      ```text
+      404
+
+      There isn't a GitHub Pages site here.
+      ```
+
+      wait for 5-10 minutes and try again. It takes a while for the website to become visible.
```

Comment:
All this was not necessary for me. Webpage was directly up.

Reply:
When I tried it it took 2-3 minutes. It depends a bit on GitHub and also the nameservers used for the hostname resolution. Also changes might take some time to be visible. The documentation of GitHub Pages states the following

>  It can take up to 20 minutes for changes to your site to publish after you push the changes to GitHub. If your don't see your changes reflected in your browser after an hour, see "About Jekyll build errors for GitHub Pages sites."

[Documentation](https://docs.github.com/en/pages/getting-started-with-github-pages/creating-a-github-pages-site)